core/server/api/authentication.js

```diff
@@ -216,8 +216,8 @@ authentication = {
         }
 
         function sendResetNotification(data) {
-            var baseUrl = config.get('forceAdminSSL') ? (config.get('urlSSL') || config.get('url')) : config.get('url'),
-                resetUrl = globalUtils.url.urlJoin(baseUrl, 'ghost/reset', globalUtils.encodeBase64URLsafe(data.resetToken), '/');
+            var baseUrl = config.get('forceAdminSSL') ? globalUtils.url.urlFor('home', {secure: true}, true) : globalUtils.url.urlFor('home', true),
```

Comment:
In this case it makes sense to ask for `config.get('forceAdminSSL')`, because `/ghost` depends on the admin url! But there are cases in the code base where it does not make sense e.q. in `core/server/mail/utils.js`. It would be great to go over the cases again and verify it's correct or not.